README.md

```diff
@@ -480,40 +480,156 @@ that the wired class extends, instead of the full implementation.
+ 
+Example usage:
+ 
+```scala
+import akka.actor.{Actor, ActorRef, ActorSystem}
 
-````scala
-class A
-class MyActor(a: A) extends Actor { /* ... */ }
-
-object Example1 {
-  val a = wire[A] // compiles to: val a = new A
-  val actorSystem = ActorSystem()
-  val myActor1: ActorRef = wireAnonymousActor[MyActor]
-  // this compiles to:
-  // val myActor1: ActorRef = system.actorOf(Props(new MyActor(a))) 
+class DatabaseAccess()
+class SecurityFilter()
+class UserFinderActor(databaseAccess: DatabaseAccess, securityFilter: SecurityFilter) extends Actor {
+  override def receive: Receive = {
+    case m => // ...
+  }
 }
 
-object Example2 {
-  val a = wire[A]
-  val actorSystem = ActorSystem()
-  val myActor1: ActorRef = wireActor[MyActor]("myActorsName")
-  // this compiles to:
-  // val myActor1: ActorRef = system.actorOf(Props(new MyActor(a)), "myActorsName") 
-}
+import com.softwaremill.macwire._
+import com.softwaremill.macwire.akkasupport._
+  
+val theDatabaseAccess = wire[DatabaseAccess] //1st dependency for UserFinderActor
+                                             //it compiles to: val theDatabaseAccess = new DatabaseAccess
+					     
+val theSecurityFilter = wire[SecurityFilter] //2nd dependency for UserFinderActor 
+                                             //it compiles to: val theSecurityFilter = new SecurityFilter
+					     
+val system = ActorSystem("actor-system") //there must be instance of ActoRefFactory in scope
+
+val theUserFinder = wireActor[UserFinderActor]("userFinder")  
+//this compiles to:
+//lazy val theUserFinder = system.actorOf(Props(classOf[UserFinderActor], theDatabaseAccess, theSecurityFilter), "userFinder")
+``` 
+
+Note how easy is to create actorRef:
+
+```scala
+wireActor[UserFinderActor]("userFinder")
+```
+ 
+compared to:
+ 
+```scala
+system.actorOf(Props(classOf[UserFinderActor], theDatabaseAccess, theSecurityFilter), "userFinder")
+```
+
+In order to make it working all dependencies created `Actor`'s (`UserFinderActor`'s) primary constructor and 
+instance of the `akka.actor.ActorRefFactory` must be in scope. In above example this is all true. Dependencies
+of the `UserFinderActor` are `DatabaseAccess` and `SecurityFilter` and they are in scope. 
+The `ActorRefFactory` is in scope as well because `ActorSystem` which is subtype of it is there.
+
+
+Using Macwire for wiring Actors can be attractive not only because it safes from to much typing. 
+It is as well type safer because possible errors will be reported by scala compiler. In standard way 
+of creating actors if parameters passed to the `Props` are mistaken
+error happens during runtime. However relying on `wireActor[A]` here compilation error is raised. 
+
+Although this code is wrong it will compile and throw exception during runtime:
+```scala
+system.actorOf(
+  Props(classOf[UserFinderActor], theDatabaseAccess),  //no theSecurityFilter 
+  "userFinder"
+)
+```
```

Comment:
As I've explained, this is not the best practice in any case so why refer to it?